docs/actions.md

```diff
@@ -1,182 +1,129 @@
+Use thunks to defer state updates, create [getters](#getters), scoped mixins, etc.
+
+## Async Updates
+
+Use [thunks](#thunks) to update the state asynchronously, e.g., after a [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) resolves.
+
+[Try it Online](https://codepen.io/hyperapp/pen/ZeByKv?editors=0010)
 
 ```jsx
 app({
-  state: 0,
-  view: (state, actions) =>
-    <main>
-      <h1>
-        {state}
-      </h1>
-      <button onclick={actions.upLater}>+1</button>
-    </main>,
   actions: {
-    upLater: (state, actions) =>
-      new Promise(resolve => setTimeout(resolve, 1000, state + 1))
+    getURL(state) {
+      return update => fetch(`/search?q=${state.query}`)
+        .then(data => data.json())
+        .then(json => update({
+          url: json[0].url
+        })
+      )
+    }
   }
 })
 ```
 
-Actions can be written as [async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) too.
+Actions need not have a return value at all. You can use them to create side effects, call other actions, etc.
 
 ```jsx
-const delay = result =>
-  new Promise(resolve => setTimeout(resolve, 1000, result))
-
 app({
-  state: 0,
-  view: (state, actions) =>
-    <main>
-      <h1>
-        {state}
-      </h1>
-      <button onclick={actions.upLater}>+1</button>
-    </main>,
   actions: {
-    upLater: async state => await delay(state + 1)
+    setURL(state, actions, data) {
+      return { url: data[0].url }
+    },
+    getURL(state, actions) {
+      const req = new XMLHttpRequest()
+
+      req.open("GET", `/search?q=${state.query}`)
+      req.onreadystatechange = () => {
+        if (
+          req.readyState === XMLHttpRequest.DONE &&
+          req.status === 200
+        ) {
+          actions.setURL(JSON.parse(req.responseText))
+        }
+      }
+      req.send()
+    }
   }
 })
 ```
 
-## Namespaces
+## Getters
 
-Namespaces let you organize actions into categories or domains.
+A getter is an action that retrieves a property from the state tree or the result of a computation.
 
 ```jsx
 app({
-  state: 0,
-  view: (state, actions) =>
-    <main>
-      <button onclick={actions.counter.up}>+</button>
-      <h1>
-        {state}
-      </h1>
-      <button onclick={actions.counter.down}>-</button>
-    </main>,
   actions: {
-    counter: {
-      up: state => state + 1,
-      down: state => state - 1
+    isAdult({ userId }) {
+      return () => state.users[userId].age >= state.adultAge
     }
   }
 })
 ```
 
-## Complex State
-
-Suppose we have a complex state object and wish to update a given property avoiding mutation.
-
-Here is one way we could achieve this using [Ramda](https://github.com/ramda/ramda).
+## Namespaces
 
-[Try it online](https://codepen.io/hyperapp/pen/Zygvbg?editors=0010)
+We iterate over your action keys, allowing for namespaced actions. Namespaces help you organize your actions into categories or domains.
```

Comment:
Hyperapp iterates over action keys recursively during setup, this means that actions can be _nested_ under developer defined namespaces.